### PR TITLE
Create a guide on how to connect an app to a database hosted on the cloud

### PR DESCRIPTION
**Pull request by freddyaboulton**
# Description

Part of #2049 

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My code follows the style guidelines of this project
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: guides/5)other_tutorials/connecting_to_a_database.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+1. Are electric bikes more popular than regular bikes?
+2. What are the top 5 most popular departure bike stations?
+
+
+## Step 1 - Creating your database
+
+We will be storing our data on a PostgreSQL hosted on Amazon's RDS service. Create an AWS account if you don't already have one
+and create a PostgreSQL database on the free tier. 
+
+**Important**: If you plan to host this demo on HuggingFace spaces, make sure database is on port **8080**. Spaces will
```
Comment (abidlabs) [inline guides/5)other_tutorials/connecting_to_a_database.md:28]: Good catch!

File: guides/5)other_tutorials/connecting_to_a_database.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+```bash
+DB_USER='username' DB_PASSWORD='password' DB_HOST='host' python app.py
+```
+
+## Step 2.b - Write your ETL code
+We will be querying our database for the total count of rides split by the type of bicycle (electric, standard, or docked).
+We will also query for the total count of rides that depart from each station and take the top 5. 
+
+We will then take the result of our queries and visualize them in with matplotlib.
```
Comment (abidlabs) [inline guides/5)other_tutorials/connecting_to_a_database.md:63]: I think `plotly` plots are much nicer and more interactive than `matplotlib` plots, and just generally make for a nicer dashboard experience. What do you think?

File: guides/5)other_tutorials/connecting_to_a_database.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        [s[1] for s in stations], rotation=45, ha="right", rotation_mode="anchor"
+    )
+    ax.tick_params(axis="x", labelsize=8)
+    fig_m.tight_layout()
+    return fig_m
+```
+
+## Step 2.c - Write your gradio app
+We will display or matplotlib plots in two separate `gr.Plot` components. We will fetch the latest data from the
+database each time the application loads.
```
Comment (abidlabs) [inline guides/5)other_tutorials/connecting_to_a_database.md:116]:
```suggestion
database each time the web page loads.
```

File: guides/5)other_tutorials/connecting_to_a_database.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+    demo.load(get_count_ride_type, inputs=None, outputs=bike_type)
+    demo.load(get_most_popular_stations, inputs=None, outputs=station)
+
+demo.launch()
+```
+
+## Step 3 - Deployment
+We will be deploying our app to HuggingFace spaces. If you wish, you could also run your app locally and get a
+shareable link by passing the `share=True` parameter to `launch`.
```
Comment (abidlabs) [inline guides/5)other_tutorials/connecting_to_a_database.md:132]: Suggest rewording a little bit:
```suggestion
If you run the code above, your app will start running locally. You can even get a temporary shareable link by passing the `share=True` parameter to `launch`.

But what if you want to a permanent deployment solution? Let's deploy our Gradio app to the free HuggingFace Spaces platform.
```

**Comment by abidlabs** [comment]
Looks fantastic @freddyaboulton! Sprinked a few suggestions to consider

**Review by abidlabs — APPROVED** [review]

File: guides/5)other_tutorials/connecting_to_a_database.md (hunk 2 above)
Comment (freddyaboulton) [reply guides/5)other_tutorials/connecting_to_a_database.md:63]: Let me keep matplotlib plot for now. I tried switching over to plotly and I can't get the output I want. Can't tell if it's a bug or me 😂

File: guides/5)other_tutorials/connecting_to_a_database.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+to your data, you can display it in a web UI with gradio 💪
+
+## Overview 
+    
+We will be analyzing bike share data from Chicago. The data is hosted on kaggle [here](https://www.kaggle.com/datasets/evangower/cyclistic-bike-share?select=202203-divvy-tripdata.csv).
+Our goal is to create a dashboard that will enable our business stakeholders to answer the following questions:
+
+1. Are electric bikes more popular than regular bikes?
+2. What are the top 5 most popular departure bike stations?
+
```
[the remark this answers is not in the document]
Comment (freddyaboulton) [reply guides/5)other_tutorials/connecting_to_a_database.md:21]: Great idea!

**Comment by freddyaboulton** [comment]
Thank you for the great feedback @abidlabs !
